diff --git a/tdriver/Makefile b/tdriver/Makefile
index 9f6a665..eaf5756 100644
--- a/tdriver/Makefile
+++ b/tdriver/Makefile
@@ -1,7 +1,7 @@
 all: tdriver
 
 tdriver: main.c
-	cc -o tdriver main.c -lrt
+	cc -o tdriver main.c
 
 .PHONY: clean
 clean:
diff --git a/tdriver/main.c b/tdriver/main.c
index 734c3fe..1b1e78d 100644
--- a/tdriver/main.c
+++ b/tdriver/main.c
@@ -16,6 +16,7 @@
 #define BUFLEN	1024
 #define DEVICE	"/dev/modem"
 #define GIGA	1000000000
+#define MEGA	1000000
 #define SPEED	"300"
 
 #define MAX(a, b)	((a) > (b) ? (a) : (b))
@@ -37,7 +38,7 @@ handler(int sig)
 int
 main(int argc, char *argv[])
 {
-	struct itimerspec it;
+	struct itimerval it;
 	struct sigaction sa;
 	struct sigevent sigev;
 	struct termios tos;
@@ -45,7 +46,6 @@ main(int argc, char *argv[])
 	fd_set rfds;
 	size_t buflen;
 	ssize_t bytes;
-	timer_t timerid;
 	unsigned long period, pnprev, pnsec;
 	int c, dev, fd, i, j, nfds, psec, reqSpeed;
 	char *buf, *device, *endp, *filename, *speed;
@@ -83,26 +83,23 @@ main(int argc, char *argv[])
 	/* シグナルハンドラの設定 */
 	sa.sa_handler = handler;
 	(void)sigemptyset(&sa.sa_mask);
-	if (sigaction(SIGRTMIN, &sa, NULL) == -1)
+	if (sigaction(SIGALRM, &sa, NULL) == -1)
 		err(EXIT_FAILURE, "sigaction");
 
-	/* タイマを作る */
-	sigev.sigev_notify = SIGEV_SIGNAL;
-	sigev.sigev_signo = SIGRTMIN;
-	if (timer_create(CLOCK_MONOTONIC, &sigev, &timerid) == -1)
-		err(EXIT_FAILURE, "timer_create");
 	/* 16 周期分のタイマ時間を用意しておく */
-	period = GIGA / strtoul(speed, NULL, 0);
+	period = MEGA / strtoul(speed, NULL, 0);
 	pnprev = pnsec = psec = 0;
 	for (i = 0; i < 16; i++) {
-		pnsec = (pnprev + period) % GIGA;
+		pnsec = (pnprev + period) % MEGA;
 		if (pnsec < pnprev)
 			psec++;
 		pnprev = pnsec;
 	}
-	it.it_interval.tv_sec = it.it_interval.tv_nsec = 0;
-	it.it_value.tv_sec = period / GIGA * 16 + psec;
-	it.it_value.tv_nsec = pnsec;
+	it.it_interval.tv_sec = it.it_interval.tv_usec = 0;
+	it.it_value.tv_sec = period / MEGA * 16 + psec;
+	it.it_value.tv_usec = pnsec;
+	/* 以降のために period を合わせておく */
+	period = GIGA / strtoul(speed, NULL, 0);
 
 	/* 送信機を開く */
 	dev = open(device, O_RDWR | O_NOCTTY);
@@ -150,8 +147,8 @@ main(int argc, char *argv[])
 				if (strncmp(buf, "?", bytes) != 0)
 					continue;
 				/* 受信機が全てを忘却するまで待つ */
-				if (timer_settime(timerid, 0, &it, NULL) == -1)
-					err(EXIT_FAILURE, "timer_settime");
+				if (setitimer(ITIMER_REAL, &it, NULL) == -1)
+					err(EXIT_FAILURE, "setitimer");
 				(void)pause();
 				/* 速度を再度送信する */
 				reqSpeed = 1;
